examples/nestjs/consumer/test/app.spec.ts

```diff
@@ -0,0 +1,267 @@
+  // use unit-style tests that test the collaborating function behaviour -
+  // we want to test the function that is calling the external service.
+  describe('when a call to list all animals from the Animal Service is made', () => {
+    describe('and the user is not authenticated', () => {
+      beforeAll(() =>
+        provider.addInteraction({
+          state: 'is not authenticated',
+          uponReceiving: 'a request for all animals',
+          withRequest: {
+            method: 'GET',
+            path: '/animals/available',
+          },
+          willRespondWith: {
+            status: HttpStatus.UNAUTHORIZED,
+          },
+        }),
+      );
+
+      it('returns a 401 unauthorized', () => {
+        return expect(animalsService.suggestion(suitor)).rejects.toThrowError();
+      });
+    });
+
+    describe('and the user is authenticated', () => {
+      describe('and there are animals in the database', () => {
+        beforeAll(() =>
+          provider.addInteraction({
+            state: 'Has some animals',
+            uponReceiving: 'a request for all animals',
+            withRequest: {
+              method: HTTPMethod.GET,
+              path: '/animals/available',
+              headers: { Authorization: 'Bearer token' },
+            },
+            willRespondWith: {
+              status: HttpStatus.OK,
+              headers: {
+                'Content-Type': 'application/json; charset=utf-8',
+              },
+              body: animalListExpectation,
+            },
+          }),
+        );
+
+        it('returns a list of animals', async () => {
+          const suggestedMates = await animalsService.suggestion(suitor);
+
+          expect(suggestedMates).toHaveProperty('suggestions');
+
+          const { suggestions } = suggestedMates;
+
+          expect(suggestions).toHaveLength(MIN_ANIMALS);
+          expect(suggestions[0].score).toBe(94);
+        });
+      });
+    });
+  });
+
+  describe('when a call to the Animal Service is made to retreive a single animal by ID', () => {
+    describe('and there is an animal in the DB with ID 1', () => {
+      beforeAll(() =>
+        provider.addInteraction({
+          state: 'Has an animal with ID 1',
+          uponReceiving: 'a request for an animal with ID 1',
+          withRequest: {
+            method: HTTPMethod.GET,
+            path: term({ generate: '/animals/1', matcher: '/animals/[0-9]+' }),
+            headers: { Authorization: 'Bearer token' },
+          },
+          willRespondWith: {
+            status: HttpStatus.OK,
+            headers: {
+              'Content-Type': 'application/json; charset=utf-8',
+            },
+            body: animalBodyExpectation,
+          },
+        }),
+      );
+
+      it('returns the animal', async () => {
+        const suggestedMates = await animalsService.getAnimalById(11);
+
+        expect(suggestedMates).toHaveProperty('id', 1);
```

Comment:
You could improve this by using pact's `extractPayload` to strip the example from the matchers:

```
expect(suggestedMates).toEqual(extractPayload(animalBodyExpectation))
```

Also, a little off topic, but I don't think `suggestedMates` is the intended name for this variable - maybe `animal` or `retrievedAnimal`?

Reply:
I resolved the `extractPayload`, but didn't rename the variable